zavod/zavod/runner/local_enricher.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from typing import Generator, List, Optional
+from followthemoney.namespace import Namespace
+from tempfile import mkdtemp
+from followthemoney.types import registry
+
+from nomenklatura.entity import CE
+from nomenklatura.dataset import DS
+from nomenklatura.cache import Cache
+from nomenklatura.enrich.common import Enricher, EnricherConfig
+from nomenklatura.enrich.common import EnrichmentException
+from nomenklatura.index.tantivy_index import TantivyIndex
+from nomenklatura.matching import get_algorithm
+
+from zavod.archive import dataset_state_path
+from zavod.meta import get_catalog
+from zavod.store import get_store, get_view
+
+
+log = logging.getLogger(__name__)
+
+MATCH_CANDIDATES = 10
+
+
+class LocalEnricher(Enricher):
+    """
+    Uses a local index to look up entities in a given dataset.
+
+    Candidates are selected for matching based on the number of tokens
+    they share with the entity being matched. Candidates are then scored
+    by the matching algorithm to determine if they are a match.
+
+    Configuration:
+        `config.dataset`: `str` - the name of the dataset to enrich against.
+        `config.threshold`: `float` - the threshold to be considered a match
+            according to the matching algorithm used.
+        `config.algorithm`: `str` (default logic-v1) - the name of the algorithm
+            to use for matching.
+    """
+
+    def __init__(self, dataset: DS, cache: Cache, config: EnricherConfig):
+        super().__init__(dataset, cache, config)
+        target_dataset_name = config.pop("dataset")
+        target_dataset = get_catalog().require(target_dataset_name)
+        self._view = get_view(target_dataset, external=False)
+        state_path = dataset_state_path(dataset.name)
+        self._index = TantivyIndex(
+            self._view, state_path, dataset.config.get("index_options", {})
+        )
+        self._index.build()
+        algo_name = config.pop("algorithm", "logic-v1")
+        self._algorithm = get_algorithm(algo_name)
+        if self._algorithm is None:
+            raise EnrichmentException(f"Unknown algorithm: {algo_name}")
+        self._threshold = config.pop("threshold")
+        self._ns: Optional[Namespace] = None
+        if self.get_config_bool("strip_namespace"):
+            self._ns = Namespace()
+
+    def match(self, entity: CE) -> Generator[CE, None, None]:
```

Comment:
@pudo is it fair to change the `nomenklatura.enricher.Enrich` interface to say this yields a `nomenklatura.entity.CompositeEntity`, but not necessarily `CE`? Since this enricher will only ever yield a `zavod.entity.Enricher` regardless of what's passed into `match`

```suggestion
    def match(self, entity: CE) -> Generator[CE, None, None]:
```

Reply:
ahhh that's the sort of trick I was looking for. I knew I'd seen it somewhere. Thanks!